flink-clients/src/main/java/org/apache/flink/client/program/PackagedProgram.java

```diff
@@ -184,8 +184,6 @@ public PackagedProgram(File jarFile, List<URL> classpaths, @Nullable String entr
 			}
 
 			checkJarFile(jarFileUrl);
-		} else if (!isPython) {
-			throw new IllegalArgumentException("The jar file must not be null.");
```

Comment:
Do you think it makes sense that I open a Jira to following this issue? Because this change involves many other components.

Reply:
If to do the rework in another ticket and keep the constructor public in this PR, I think a check is needed to ensure the `entryPointClassName` and `jarFile` are not null at the same time, otherwise an NPE would happen, as till mentioned.

Reply:
Yeah, we can do this as a follow up but then we need to add this check as @zhuzhurk suggested. On the other hand, I think this change would not be terribly big.

Reply:
I checked the code again and indeed it is not so big. I think I could finish the rework in this pr.